src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacKeyFieldDeclaration/KeyFieldDeclarationFix.cs

```diff
@@ -0,0 +1,142 @@
+
+			string codeActionBoundKeysName = nameof(Resources.PX1055Fix2).GetLocalized().ToString();
+			CodeAction codeActionBoundKeys = CodeAction.Create(codeActionBoundKeysName,
+																cToken => RemoveKeysFromFieldsAsync(document,
+																									cToken,
+																									diagnostic,
+																									CodeFixModes.EditIdentityAttribute),
+																equivalenceKey: codeActionBoundKeysName);
+
+			string codeActionRemoveIdentityColumnName = nameof(Resources.PX1055Fix3).GetLocalized().ToString();
+			CodeAction codeActionRemoveIdentityColumn = CodeAction.Create(codeActionRemoveIdentityColumnName,
+																			cToken => RemoveKeysFromFieldsAsync(document,
+																												cToken,
+																												diagnostic,
+																												CodeFixModes.RemoveIdentityAttribute),
+																			equivalenceKey: codeActionRemoveIdentityColumnName);
+
+			context.RegisterCodeFix(codeActionIdentityKey, context.Diagnostics);
+			context.RegisterCodeFix(codeActionBoundKeys, context.Diagnostics);
+			context.RegisterCodeFix(codeActionRemoveIdentityColumn, context.Diagnostics);
+
+			return Task.FromResult(true);
+		}
+
+		private async Task<Document> RemoveKeysFromFieldsAsync(Document document,
+																CancellationToken cancellationToken,
+																Diagnostic diagnostic,
+																CodeFixModes mode)
+		{
+
+			SemanticModel semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+			cancellationToken.ThrowIfCancellationRequested();
+
+			Location[] attributeLocations = diagnostic.AdditionalLocations.ToArray();
+
+			SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+			List<SyntaxNode> deletedNodes = new List<SyntaxNode>();
+
+			foreach (var attributeLocation in attributeLocations)
+			{
+
+				if (!(root?.FindNode(attributeLocation.SourceSpan) is AttributeSyntax attributeNode))
+					return document;
+
+				ITypeSymbol attributeType = semanticModel.GetTypeInfo(attributeNode, cancellationToken).Type;
+
+				if (attributeType == null)
+					return document;
+
+				var attributeInformation = new AttributeInformation(new PXContext(semanticModel.Compilation));
+				bool isIdentityAttribute = attributeInformation.IsAttributeDerivedFromClass(attributeType, new PXContext(semanticModel.Compilation).FieldAttributes.PXDBIdentityAttribute) ||
+										   attributeInformation.IsAttributeDerivedFromClass(attributeType, new PXContext(semanticModel.Compilation).FieldAttributes.PXDBLongIdentityAttribute);
+
+
+				if ((mode == CodeFixModes.EditIdentityAttribute && isIdentityAttribute) ||
+					(mode == CodeFixModes.EditKeyFieldAttributes && !isIdentityAttribute))
+				{
+					IEnumerable<AttributeArgumentSyntax> deletedNode = GetIsKeyEQTrueArguments(attributeNode);
+
+					deletedNodes.AddRange(deletedNode);
+				}
+				if (mode == CodeFixModes.RemoveIdentityAttribute && isIdentityAttribute)
```

Comment:
@saloponov maybe `else` is missed here. If not then please put empty lines to separate control flow statements